### PR TITLE
More Semigroup additions

### PR DESCRIPTION
**Pull request by puffnfresh**
1. dual
2. min/max
3. lift
4. sumRight

### CODE REVIEW

File: core/src/main/java/fj/Semigroup.java [hunk 1]
```diff
@@ -97,6 +98,27 @@ public A multiply1p(int n, A a) {
   }
 
   /**
+   * Sums the given values with right-fold.
+   */
+  public A sumRight(final NonEmptyList<A> as) {
```
Comment (jbgi) [inline core/src/main/java/fj/Semigroup.java]: what about naming it ```sumNel```? associativity make sumRight and sumLeft equivalent, and thus using foldLeft1 would be more performant...

Comment (jbgi) [inline core/src/main/java/fj/Semigroup.java]: @puffnfresh WDYT?

Comment (puffnfresh) [inline core/src/main/java/fj/Semigroup.java]: @jbgi I agree. I will update this.

Comment (jbgi) [inline core/src/main/java/fj/Semigroup.java]: sorry, I did a merge of my branch and you will need to rebase now...

**Comment by puffnfresh** [comment]
@jbgi rebased and fixed!

**Pull request merged**